src/Mantasflowers.Contracts/Payment/Request/PostCreateCouponRequest.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Newtonsoft.Json;
+using System;
+
+namespace Mantasflowers.Contracts.Payment.Request
+{
+    public class PostCreateCouponRequest
+    {
+        public string Name { get; set; }
+
+        public long DurationInMonths { get; set; }
+
+        [JsonIgnore]
+        public DateTime BeginDate { get; set; }
+
+        public DateTime RedeemBy { get; set; }
+
+        public long DiscountPrice { get; set; }
+
+        public long OrderOverPrice { get; set; }
```

Comment:
These are decimal in DB. Also don't remember if `DiscountPrice` is the percent or what (weird naming on our part?).

Reply:
Yeah, the thing is that stripe only supports long. DiscountPrice was weird on my part too. The idea was to discount specific amount rather than percentage.

Reply:
Its just if this is specific discount (i.e., price after discount), it almost makes it so that we have 1 coupon per 1 product

Reply:
You can specify for which products discount is applicable, however, currently it should be deducted from the total.

Reply:
DiscountPrice doesn't sound like something you would deduce from the actual price. We'll have to keep this in mind, because it's already getting confusing :d